ui/src/app/components/rollouts-toolbar/rollouts-toolbar.tsx

```diff
@@ -0,0 +1,230 @@
+        showRequiresAttention: searchParams.get('showRequiresAttention') === 'true',
+        showFavorites: searchParams.get('showFavorites') === 'true',
+        name: searchParams.get('name') || '',
+        displayMode: searchParams.get('displayMode') || defaultDisplayMode,
+        status: {
+            Progressing: searchParams.get('Progressing') === 'true',
+            Degraded: searchParams.get('Degraded') === 'true',
+            Paused: searchParams.get('Paused') === 'true',
+            Healthy: searchParams.get('Healthy') === 'true',
+        },
+    });
+    // Ensure that the filters are updated when the URL changes
+    onFilterChange(filters);
+
+    const handleFilterChange = (newFilters: Filters) => {
+        setFilters(newFilters);
+        onFilterChange(newFilters);
+    };
+
+    const handleNameFilterChange = (value: string) => {
+        const newFilters = {
+            ...filters,
+            name: value,
+        };
+        const searchParams = new URLSearchParams(location.search);
+        if (value) {
+            searchParams.set('name', value);
+        } else {
+            searchParams.delete('name');
+        }
+        history.push({search: searchParams.toString()});
+        handleFilterChange(newFilters);
+    };
+
+    const handleShowRequiresAttentionChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+        const newFilters = {
+            ...filters,
+            showRequiresAttention: !filters.showRequiresAttention,
+        };
+        const searchParams = new URLSearchParams(location.search);
+        if (!filters.showRequiresAttention) {
+            searchParams.set('showRequiresAttention', 'true');
+        } else {
+            searchParams.delete('showRequiresAttention');
+        }
+        history.push({search: searchParams.toString()});
+        handleFilterChange(newFilters);
+    };
+
+    const handleShowFavoritesChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+        const newFilters = {
+            ...filters,
+            showFavorites: !filters.showFavorites,
+        };
+        const searchParams = new URLSearchParams(location.search);
+        if (!filters.showFavorites) {
+            searchParams.set('showFavorites', 'true');
+        } else {
+            searchParams.delete('showFavorites');
+        }
+        history.push({search: searchParams.toString()});
+        handleFilterChange(newFilters);
+    };
+
+    const handleDisplayModeChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+        const newFilters = {
+            ...filters,
+            displayMode: event.currentTarget.id,
+        };
+        const searchParams = new URLSearchParams(location.search);
+        if (event.currentTarget.id !== defaultDisplayMode) {
+            searchParams.set('displayMode', event.currentTarget.id);
+        } else {
+            searchParams.delete('displayMode');
+        }
+        history.push({search: searchParams.toString()});
```

Comment:
what is the reason to update the url params, can you manage the filter by using state

Reply:
Updating the urlparams allows us to share URLs, deep-linking to a specific filter